### PR TITLE
Bring back some deprecated utilities

### PR DESCRIPTION
**Pull request by thecrypticace**
These utilities are deprecated (and were removed) but we're brining them back so they keep working:

- `flex-grow-*`
- `flex-shrink-*`
- `decoration-slice`
- `decoration-clone`
- `overflow-ellipsis`

### CODE REVIEW

File: packages/tailwindcss/src/utilities.ts [hunk 1]
```diff
@@ -1929,6 +1965,10 @@ export function createUtilities(theme: Theme) {
   ])
 
   staticUtility('text-ellipsis', [['text-overflow', 'ellipsis']])
+
+  /** @deprecated */
+  staticUtility('overflow-ellipsis', [['text-overflow', 'ellipsis']])
```
Comment (philipp-spiess) [inline packages/tailwindcss/src/utilities.ts:1970]: We should add this here: https://github.com/tailwindlabs/tailwindcss/blob/next/packages/tailwindcss/src/compat/legacy-utilities.ts

Comment (thecrypticace) [reply packages/tailwindcss/src/utilities.ts:1970]: ohhh I forgot we'd split these. Thanks for the reminder!

File: packages/tailwindcss/src/compat/legacy-utilities.test.ts [hunk 2]
```diff
@@ -131,3 +131,83 @@ test('max-w-screen', async () => {
...
+      -webkit-box-decoration-break: slice;
+      box-decoration-break: slice;
+    }"
+  `)
+})
+
+test('overflow-ellipsis', async () => {
+  expect(await run(['overflow-ellipsis'])).toMatchInlineSnapshot(`
+    ".overflow-ellipsis {
+      text-overflow: linear-gradient(var(--tw-gradient-stops));
```
Comment (philipp-spiess) [inline packages/tailwindcss/src/compat/legacy-utilities.test.ts:152]: This looks wrong? 🤔

Comment (thecrypticace) [reply packages/tailwindcss/src/compat/legacy-utilities.test.ts:152]: lol um

Comment (thecrypticace) [reply packages/tailwindcss/src/compat/legacy-utilities.test.ts]: fixed

**Review by philipp-spiess — APPROVED** [review]

**Pull request merged**